sdk/core/core/src/az_span.c

```diff
@@ -52,13 +52,14 @@ AZ_NODISCARD az_span az_span_from_str(char* str)
 
 AZ_NODISCARD az_span az_span_slice(az_span span, int32_t start_index, int32_t end_index)
 {
-  AZ_PRECONDITION_VALID_SPAN(span, 0);
+  int32_t const span_size = az_span_size(span);
+  AZ_PRECONDITION(span_size >= 0);
```

Comment:
This should just be a check that the span is valid; not just its size.
It IS OK to slice a NULL span as long as the start and end are both 0.

You should have the precondition test for start <= end first. AND, both must be >=0.
The start must also be between 0 and size+1 (I think) inclusive.